applications/istio/bookinfo/appconfig-1-bookinfo-VirtualService.yaml

```diff
@@ -0,0 +1,35 @@
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: bookinfo
+spec:
+  components:
+    - componentName: bookinfo-gateway
```

Comment:
Here you should need to reference the previously created components, for example

```yaml
    - componentName: reviews
    - componentName: details
    - componentName: ratings
    - componentName: productpage
    - componentName: reviews-v1
    - componentName: reviews-v2
    - componentName: reviews-v3
    - componentName: details-v1    
    - componentName: ratings-v1    
    - componentName: productpage-v1
    - componentName: bookinfo-gateway
```

In addition, I think service should also be used as a deployment trait, not a component